wolfcrypt/src/port/nxp/se050_port.c

```diff
@@ -1268,4 +1269,168 @@ int se050_ed25519_verify_msg(const byte* signature, word32 signatureLen,
 
 #endif /* HAVE_ED25519 */
 
+
+#ifdef HAVE_CURVE25519
+
+int se050_curve25519_create_key(curve25519_key* key, int keySize)
+{
+    sss_status_t    status = kStatus_SSS_Success;
+    sss_object_t    keyPair;
+    sss_key_store_t host_keystore;
+    int keyId = se050_allocate_key(SE050_KEYID_ANY);
+    uint8_t keyPairExport[MAX_ECC_BYTES];
+    size_t keyPairExportLen = sizeof(keyPairExport);
+    size_t keyPairExportBitLen = sizeof(keyPairExport) * 8;
+    int ret;
+
+    (void)keySize;
+
+    if (cfg_se050_i2c_pi == NULL) {
+        return WC_HW_E;
+    }
+
+    if (wolfSSL_CryptHwMutexLock() != 0) {
+        return BAD_MUTEX_E;
+    }
+    status = sss_key_store_context_init(&host_keystore, cfg_se050_i2c_pi);
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_store_allocate(&host_keystore, 60);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_object_init(&keyPair, &host_keystore);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_object_allocate_handle(&keyPair, keyId,
+            kSSS_KeyPart_Pair, kSSS_CipherType_EC_MONTGOMERY, 256,
+            kKeyObject_Mode_None);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_store_generate_key(&host_keystore, &keyPair,
+            256, NULL);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_store_get_key(&host_keystore, &keyPair,
```

Comment:
The exported public key should be placed into `key->p.point`. In fact you don't need to use the local `keyPairExport`, just sent directly to `key->p.point`. Also `key->p.pointSz = CURVE25519_KEYSIZE`.

Reply:
This seems to be done properly now at L:1330-1333